rpcs3/Emu/Cell/SPUAnalyser.cpp

```diff
@@ -5,16 +5,18 @@
 
 const spu_decoder<spu_itype> s_spu_itype;
 
-spu_function_t* SPUDatabase::find(const be_t<u32>* data, u64 key, u32 max_size)
+std::shared_ptr<spu_function_contents_t> SPUDatabase::find(const be_t<u32>* data, u64 key, u32 max_size, void* ignore)
 {
 	for (auto found = m_db.equal_range(key); found.first != found.second; found.first++)
 	{
-		const auto& func = found.first->second;
+		const auto & func = found.first->second;
+
+		// TODO remove code after a while if it hasn't been touched, else there's a big memory bloat here
 
 		// Compare binary data explicitly (TODO: optimize)
-		if (LIKELY(func->size <= max_size) && std::memcmp(func->data.data(), data, func->size) == 0)
+		if (func.get() != ignore && LIKELY(func->size <= max_size) && memcmp(func->data.data(), data, func->size) == 0)
```

Comment:
Why the hate for `std::`? rpcs3 includes `cstring` so correct namespaces should be prefered.